rfc/0001-checks-customization.md

```diff
@@ -0,0 +1,325 @@
+
+**Customizable Check excerpt**
+```yaml
+id: "156F64"
+name: Check Corosync token_timeout value
+# ...
+values:
+  - name: expected_token_timeout
+    default: 5000
+    conditions:
+      - value: 30000
+        when: env.provider == "azure" || env.provider == "aws"
+      - value: 20000
+        when: env.provider == "gcp"
+```
+
+In order to allow customization of checks where hardcoded values are used in expectation rhai scripts, those checks need to be adjusted by moving hardcoded values to DSL specified values.
+
+**Not Customizable Check excerpt**
+```yaml
+# not customizable check
+id: "AAA000"
+name: Lorem ipsum dolor sit amet
+# ...
+# missing `values` entry
+expectations:
+  - name: foo_expectation
+    expect: facts.some_fact == "yes"
+```
+
+Need to be refactored to something like the following
+```yaml
+# refactored to be customizable
+id: "AAA000"
+name: Lorem ipsum dolor sit amet
+# ...
+values:
+  - name: expected_yes_value
+    default: "yes"
+
+expectations:
+  - name: foo_expectation
+    expect: facts.some_fact == values.expected_yes_value
+```
+
+This is necessary because operating on the rhai script is currently not an option.
+
+### Caveats
+
+After some deeper research there are some caveates to be take into account for check customizability
+- check [3A59DC](https://github.com/trento-project/checks/blob/main/checks/3A59DC.yaml) will be not allowed for customization because of its nature (what it is attempting to do and how it is written)
+- if one of the values of a check is a list, that specific value won't be allowed for modification (deferred implementation)
+
+## Operations on custom values
+
+In order to support applying custom values, changing them after they've been set and also reset to SUSE's default ones, the following new operations would be introduced:
+- [Apply custom values](#apply-custom-values)
+- [Change custom values](#change-custom-values)
+- [Reset custom values](#reset-check-to-defaults)
+- [Reading custom values](#reading-customization)
+
+**Disclaimer:** all following endpoints, methods, paths, query strings, parameters are indicative at this point and up for further refinement.
+The intent is to give the general feeling of the moving parts and what is needed.
+
+### Apply custom values
+This operation allows to apply custom values for a given customizable check that has not been customized yet.
+The operation allows to apply a custom value for at least one customizable value, meaning that if a check defines more than one **expected value**, this operation allows to customize one, many or all of them (ie. if a check has 3 expected values defined, a user might be interested in customizing only one of them and use the default ones for the others)
+
+#### Endpoint
+
+`POST /checks/:check_id/values`
+```json
+{
+    "target_id": "target-uuid",
```

Comment:
What are this `target_id` and `group_id`?

Reply:
target_id is basically a host id and group_id is either a cluster id or a host id depending on whether we're running checks for a host or for a cluster. 

Wanda uses those.

Reply:
nope. a customized value for a cluster is applied to all the hosts of that target